force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -102,18 +123,18 @@ public with sharing class ClassesObjectsMethods {
      */
     public static Integer countVowels(String str) {
         // Initialize the result Integer
-
-        /*
-        for () {
-            // get individual characters from the string - hint use substring or split
-            // check if the character is a vowel
-            // if it is a vowel, increment the result Integer
-        }
-        */
-
-        return null; // Replace null with the variable you used to store the result
+        String validCharString = 'aeiou';
+        Integer totalVowels = 0;
+
+        for (integer i=0; i < str.length(); i++){
+            if(validCharString.contains((str.mid(i,1)).toLowerCase())) {
+                totalVowels++;
+            }
+        }        
+        System.debug('totalVowels=' + totalVowels); 
+        return totalVowels;
     }
```

Comment:
I would declare a set to store my vowels. Much more more performant than a String.  My solution is below. But this is good work. I don't think I realized there was mid method. I just had to look it up. 

```
 public static Integer countVowels(String str) {
        // Initialize the result Integer
        Integer vowelCount = 0;
        Set<String> vowels = new Set<String>{
                'a', 'e', 'i', 'o', 'u'
        };
        List<String> characters = str.split('');
        for (String character : characters) {
            if (vowels.contains(character.toLowerCase())) {
                vowelCount++;
            }
        }
        return vowelCount;
    }
    ```